### PR TITLE
docs: updating the documentation for Extension Servers and adding an example extension server

### LINKED ISSUE #1726: Demonstrate how a user would use the Extension Manager Framework with example
*Description*:

Add documentation highlighting how a user would go about leveraging the extension manager framework https://github.com/envoyproxy/gateway/blob/main/docs/latest/design/extending-envoy-gateway.md to implement some use case

### PR DESCRIPTION
**Pull request by liorokman**
**What this PR does / why we need it**:
This PR updates the Envoy Extension Server documentation and adds an example extension server.

**Which issue(s) this PR fixes**:
Fixes #1726  
Replaces #2931

### CODE REVIEW

**Comment by liorokman** [comment]
@AliceProxy  please take a look

**Comment by Xunzhuo** [comment]
Hope that we can get this into v1.1

File: site/content/en/contributions/design/extending-envoy-gateway.md [hunk 1]
```diff
@@ -53,6 +53,10 @@ An example configuration:
 apiVersion: gateway.envoyproxy.io/v1alpha1
 kind: EnvoyGateway
 extensionManager:
+  poliyResources:
```
Comment (AliceProxy) [inline site/content/en/contributions/design/extending-envoy-gateway.md:56]: `poliyResources` -> `policyResources`

File: site/content/en/latest/tasks/extensibility/extension-server.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+programmatically using any semantics supported by the [xDS][] API.
+
+Using an extension server allows vendors to add xDS configuration that Envoy Gateway itself
+doesn't support with a very high level of control over the generated xDS configuration.
+
+**Note:** Modifying the xDS configuration generated by Envoy Gateway may break functionality
+configured by native Envoy Gateway means. Like other cases where the xDS configuration
+is modified outside of Envoy Gateway's control, this is risky and should be tested thoroughly,
+especially when using the same extension server across different Envoy Gateway versions.
+
```
Comment (AliceProxy) [inline site/content/en/latest/tasks/extensibility/extension-server.md:17]: not essential, but I would add another note here about Envoy Gateway only supporting a single extension at any given time.

**Review by AliceProxy — APPROVED** [review]

**Review by guydc — APPROVED** [review]

**Pull request merged**